api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -205,6 +205,24 @@ message ExternalProcessor {
 
   // Options related to the sending and receiving of dynamic metadata.
   MetadataOptions metadata_options = 16;
+
+  // If true, send each part of the HTTP request or response specified by ProcessingMode
+  // without pausing on filter chain iteration. It is "Send and Go" mode that can be used
+  // by external processor to observe Envoy data and status. In this mode:
+  // 1) Only STREAMED body processing mode is supported and any other body processing modes
+  // will be ignored.
+  // 2) External processor is not expected to send back processing response. Any response
+  // will be simply ignored.
```

Comment:
Are we removing the `async` field? If not, I'm not sure what the distinction is. I'm also not sure if `observability_mode` is any clearer.

Reply:
The existing async field is unused though, so this is not technically overloading.

Reply:
I meant to say reuse. `async_mode` seems like a poor name for a read-only mirror mode.

Reply:
I agreed that read-only is a bit implicit in `async_mode`, though I was hoping code comment can be used to clarify this.

IIUC about the discussion, the deciding factor here actually is: Whether we can reuse `#not-implemented-hide` field that might be confusing and has not been implemented for a long time:

- On one hand, reusing this field could cause "surprise"/problem to the customers who configure this `#not-implemented-hide` field by mistake or by misunderstanding of this field, as my implementation will introduce a new behavior. Should Envoy be responsible for this?

- On the other hand, not reusing this field essentially seems to indicate `#not-implemented-hide` is useless due to above concern. Besides, the case here for example, leaving `async_mode` there while adding a new config probably also not good for code health. Can we remove the old field if we want to add a similar new field?


I think this probably becomes a bigger question to @envoyproxy/api-shepherds and cc @yanavlasov @htuch @adisuissa  (though it might be slightly outside scope of this PR :) ).

Reply:
I would just mark it as `reserved` in this case to prevent the field number confusion issue and move on given the `#not-implemented-hide` status.

Reply:
Done！